### PR TITLE
region_cache: fix bug when reload tiflash_compute store cache

### PR DESCRIPTION
**Pull request by guo-shaoge**
1. fix bug in `reloadTiFlashComputeStores`, should only reload store whose state is up.
2. change all tiflash_mpp to tiflash_compute

### CODE REVIEW

**Comment by disksing** [comment]
@guo-shaoge Please fix linter issue. You can see the detail in CI log.

**Comment by guo-shaoge** [comment]
Actually I only changed 3 files, other 10 files are changed by gofmt. Don't know why other 10 files need change.

**Comment by sticnarf** [comment]
> Actually I only changed 3 files, other 10 files are changed by gofmt. Don't know why other 10 files need change. @disksing PTAL~

I guess you're formatting the whole project with Go 1.19.

Go 1.19 has different fmt rules from 1.18. It enforces the format of comments. Let's keep them unchanged for now.

**Comment by guo-shaoge** [comment]
> > Actually I only changed 3 files, other 10 files are changed by gofmt. Don't know why other 10 files need change. @disksing PTAL~
> 
> I guess you're formatting the whole project with Go 1.19.
> 
> Go 1.19 has different fmt rules from 1.18. It enforces the format of comments. Let's keep them unchanged for now.



> > Actually I only changed 3 files, other 10 files are changed by gofmt. Don't know why other 10 files need change. @disksing PTAL~
> 
> I guess you're formatting the whole project with Go 1.19.
> 
> Go 1.19 has different fmt rules from 1.18. It enforces the format of comments. Let's keep them unchanged for now.

You mean I should not change the other 10 files? But the ci will break.

**Comment by sticnarf** [comment]
> You mean I should not change the other 10 files? But the ci will break.

Emm, that's strange. I don't see similar problems in ther PRs.

File: internal/locate/region_request.go [hunk 1]
```diff
@@ -264,8 +264,9 @@ type replicaSelector struct {
 // selectorState is the interface of states of the replicaSelector.
 // Here is the main state transition diagram:
 //
-//                                    exceeding maxReplicaAttempt
-//           +-------------------+   || RPC failure && unreachable && no forwarding
+//	                         exceeding maxReplicaAttempt
+//	+-------------------+   || RPC failure && unreachable && no forwarding
```
Comment (sticnarf) [inline internal/locate/region_request.go:268]: You can see the new formatter breaks the diagrams in the comments. At least this should be reverted.

Comment (guo-shaoge) [reply internal/locate/region_request.go]: Looks like I only need to format my 3 changed files.

File: tikvrpc/endpoint.go [hunk 2]
```diff
@@ -81,8 +81,8 @@ func GetStoreTypeByMeta(store *metapb.Store) EndpointType {
 		if label.Key == EngineLabelKey && label.Value == EngineLabelTiFlash {
 			return TiFlash
 		}
-		if label.Key == EngineLabelKey && label.Value == EngineLabelTiFlashMPP {
-			return TiFlashMPP
+		if label.Key == EngineLabelKey && label.Value == EngineLabelTiFlashCompute {
+			return TiFlashCompute
```
Comment (sticnarf) [inline tikvrpc/endpoint.go:85]: Is it backward compatible? What if such a new client connects to an old cluster with `tiflash_mpp` as its engine label?

Comment (guo-shaoge) [reply tikvrpc/endpoint.go:85]: Older version of tiflash's lable is fiexed at compilation time(can only be 'tiflash').
And in new versoin of tiflash(disaggregated tiflash mode), we ensure tiflash lable can only be 'tiflash' or 'tiflash_compute'.

Comment (guo-shaoge) [reply tikvrpc/endpoint.go:85]: e.g. 'tiflash_mpp' is never used.

Comment (sticnarf) [reply tikvrpc/endpoint.go:85]: OK.

**Review by sticnarf — APPROVED** [review]

**Pull request merged**